components/Wizard/OCIAuthStep.js

```diff
@@ -0,0 +1,156 @@
+              <label htmlFor="user-input" className="pf-l-flex__item">
+                <span className="pf-c-form__label-text">
+                  <FormattedMessage defaultMessage="User OCID" />
+                </span>
+                <span className="pf-c-form__label-required" aria-hidden="true">
+                  &#42;
+                </span>
+              </label>
+              <Popover
+                id="user-popover"
+                bodyContent={
+                  <FormattedMessage
+                    defaultMessage="You can find your user OCID {iam} page in the OCI console."
+                    values={{
+                      iam: <strong>Identity and Access Management (IAM)</strong>,
+                    }}
+                  />
+                }
+                aria-label={formatMessage(ariaLabels.user)}
+              >
+                <Button variant="plain" aria-label={formatMessage(ariaLabels.user)}>
+                  <OutlinedQuestionCircleIcon id="popover-icon" />
+                </Button>
+              </Popover>
+            </div>
+            <TextInput
+              className="pf-c-form-control"
+              value={uploadSettings.user}
+              id="user-input"
+              name="user"
+              onChange={setUploadSettings}
+            />
+          </div>
+          <div className="pf-c-form__group">
+            <div className="pf-c-form__label pf-m-no-padding-top pf-l-flex pf-u-display-flex pf-m-justify-content-flex-start pf-m-nowrap">
+              <label htmlFor="private_key-input" className="pf-l-flex__item">
+                <span className="pf-c-form__label-text">
+                  <FormattedMessage defaultMessage="Private key" />
+                </span>
+                <span className="pf-c-form__label-required" aria-hidden="true">
+                  &#42;
+                </span>
+              </label>
+              <Popover
+                id="private_key-popover"
+                bodyContent={
+                  <FormattedMessage
+                    defaultMessage="You can view your deployed RSA keys in the API Keys section on the {iam} page in the OCI console."
+                    values={{
+                      iam: <strong>Identity and Access Management (IAM)</strong>,
+                    }}
+                  />
+                }
+                aria-label={formatMessage(ariaLabels.private_key)}
+              >
+                <Button variant="plain" aria-label={formatMessage(ariaLabels.private_key)}>
+                  <OutlinedQuestionCircleIcon id="popover-icon" />
+                </Button>
+              </Popover>
+            </div>
+            <TextArea
+              className="pf-c-form-control"
+              value={uploadSettings.private_key}
+              type="text"
+              id="private_key-input"
+              name="private_key"
+              onChange={setUploadSettings}
+            />
```

Comment:
You had mentioned wanting a file upload feature here. I am bumping the patternfly version so you can now follow the documentation [here](https://www.patternfly.org/v4/components/file-upload). I would recommend not using the setUploadSettings function and instead writing a custom one specifically for the private key. You could do something like:

```
  handleFileInputChange(event, file) {
    this.setState({ filename: file.name });
  }

  handleDataChange(value) {
    this.props.setPrivateKeyFile(value);
  }

...

<FileUpload
  type="text"
  value={this.props.uploadSettings.private_key}
  filename={this.state.filename}
  filenamePlaceholder="Placeholder"
  browseButtonText="Upload"
  hideDefaultPreview
  onFileInputChange={this.handleFileInputChange}
  onDataChange={this.handleDataChange}
/>
```

Reply:
Thanks for the version bump. I'm still not sure if to stay with the textarea or the file upload. Any preferences from the whole application UX perspective?

Reply:
If this needs to be a secret value than I think using a file upload would be better. Such a large entry all as secret wouldn't be particularly easy for users I believe. If it can be plaintext than I think you could use both a text area and a file upload. If you don't include `hideDefaultPreview` the fileupload also includes a textarea.

Reply:
The patternfly packages are now updated (was waiting on tests). Feel free to rebase on top of main.

Reply:
Used it now and it works fine. I specifically set the value in the uploadSettings in the end. Do you still think a dedicated function is needed?